setup.py

```diff
@@ -47,12 +38,13 @@ def run_pack_labextension():
 
 cmdclass = create_cmdclass('pack_labext', data_files_spec=data_files_spec)
 cmdclass['pack_labext'] = combine_commands(
-    command_for_func(run_pack_labextension),
+    install_npm(Path('js'), build_cmd='build:all'),
```

Comment:
This breaks if you don't have a global `yarn` command installed (which is the case on all of my machines), which is why I was using a simple hand rolled thing that relies on `jlpm` instead. Unlike `yarn`, our build system guarantees that `jlpm` will be installed (as a part of `jupyterlab`) by the time setup.py runs

Reply:
but jlpm is just an alias for yarn?

Reply:
nope, `jlpm` is a *vended copy* of yarn with a different name. I used to do a lot of research work with the hadoop yarn scheduler, and was extremely unhappy when facebook stole their name/console script name

Reply:
but in general we should go with the one that's actually installed and set up by our build

Reply:
so then it sounds like we want the non hand rolled `install_npm` but pass in `npm='jlpm'`?
https://github.com/jupyter/jupyter-packaging/blob/master/jupyter_packaging/setupbase.py#L312

Reply:
nope, if you use `install_npm` as written your only two choices are `yarn` and `npm`. I already went through [this same nonsense with jupyterlab-git](https://github.com/jupyterlab/jupyterlab-git/blob/master/setup.py#L33). Originally in jupyterlab-git I wrote my own extended version of `install_npm`, found out that `install_npm` doesn't do anything useful anyway, and then wisely went with the dead-simple hand-rolled 5 line thing

Reply:
@vidartf owns that so if its a problem lets fix it. but it looks like it would accept jlpm: https://github.com/jupyter/jupyter-packaging/blob/master/jupyter_packaging/setupbase.py#L352

what is the error?

Reply:
Do we have something that breaks when using vanilla `npm` here?

Reply:
> pass in `npm='jlpm'`?

Ah, I see. Sorry, I missed your point there.

You can't get it to use `jlpm` by just doing `install_npm('jlpm')`, and I think at that point I decided that `install_npm` was just doing too much stuff (does it really need to do it's own stale check, in addition to the one that `tsc` itself does?), so I went with something simple.

If you really want to use `install_npm` and `npm='jlpm'` works as intended, then that's fine with me.

Reply:
> does it really need to do it's own stale check, in addition to the one that tsc itself does?

It checks whether the packages in `node_modules` are stale, e.g. if you are missing `typescript`. It is hard to run `tsc` without it ;)

Reply:
In the latest setup.py I have `install_npm(js_pkg, build_cmd='build:labdist', npm=['jlpm'])`